### PR TITLE
Add additional Freedomotic unit tests

### PR DESCRIPTION
**Pull request by sudo-jaa**
Hi,

I've identified some areas in the Freedomotic codebase that could benefit from increased coverage from some additional unit tests. 
I've written the tests in this PR for a few such classes with the help of [Diffblue](https://www.diffblue.com/) [Cover](https://static1.squarespace.com/static/5bc9dd6fb2cf795fa3511f70/t/5be2e84f8985836d62e8eb13/1541597264010/Diffblue+Cover+Datasheet.pdf).

Hopefully, these tests should help you detect regressions caused by future code changes. Do let me know if you'd be interested in having any extra tests written for your project.

### CODE REVIEW

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]
Fix the commit messages.

**Review by petersear — CHANGES_REQUESTED** [review]
Tests look good to me.
The following files have a newline missing at the end of the file (I wasn't able to add comments at the relevant lines):
 framework/freedomotic-core/src/test/java/com/freedomotic/environment/GraphEdgeTest.java
 framework/freedomotic-core/src/test/java/com/freedomotic/nlp/NlpTest.java
 framework/freedomotic-core/src/test/java/com/freedomotic/rules

The order of tests in:
   framework/freedomotic-core/src/test/java/com/freedomotic/environment/GraphEdgeTest.java
is a bit random:

  public void compareToInputNotNullOutputNegative2() {
  public void compareToInputNotNullOutputPositive3() {
  public void compareToInputNotNullOutputPositive() {
  public void compareToInputNotNullOutputNegative() {
  public void compareToInputNotNullOutputPositive2() {
  public void compareToInputNotNullOutputZero() {
  public void equalsInputNotNullOutputFalse7() {
  public void equalsInputNotNullOutputFalse6() {
  public void equalsInputNotNullOutputFalse8() {
  public void equalsInputNullOutputFalse() {
  public void equalsInputNotNullOutputFalse3() {
  public void equalsInputNotNullOutputFalse4() {
  public void equalsInputNotNullOutputTrue() {
  public void equalsInputNotNullOutputFalse() {
  public void equalsInputNotNullOutputFalse2() {

I would expect them to be grouped by similar test name and in numerical order:

  public void compareToInputNotNullOutputNegative() {
  public void compareToInputNotNullOutputNegative2() {
  public void compareToInputNotNullOutputPositive() {
  public void compareToInputNotNullOutputPositive2() {
  public void compareToInputNotNullOutputPositive3() {
  public void compareToInputNotNullOutputZero() {
  public void equalsInputNotNullOutputFalse() {
  public void equalsInputNotNullOutputFalse2() {
  public void equalsInputNotNullOutputFalse3() {
  public void equalsInputNotNullOutputFalse4() {
  public void equalsInputNotNullOutputFalse6() {
  public void equalsInputNotNullOutputFalse7() {
  public void equalsInputNotNullOutputFalse8() {
  public void equalsInputNotNullOutputTrue() {
  public void equalsInputNullOutputFalse() {

**Comment by sudo-jaa** [comment]
@petersear Resolved

File: framework/freedomotic-core/src/test/java/com/freedomotic/environment/GraphEdgeTest.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+    // Act
+    final boolean retval = objectUnderTest.equals(other);
+
+    // Assert result
+    Assert.assertEquals(false, retval);
+  }
+
+  @Test
+  public void equalsInputNotNullOutputTrue() {
```
Comment (petersear) [inline framework/freedomotic-core/src/test/java/com/freedomotic/environment/GraphEdgeTest.java:206]: Alphabetically, equalsInputNotNullOutputTrue() comes before equalsInputNullOutputFalse() (NotNull before Null) but will anyone notice? I'd suggest not (well, apart from me obviously ;-)

**Review by petersear — APPROVED** [review]

File: framework/freedomotic-core/src/test/java/com/freedomotic/environment/GraphEdgeTest.java (hunk 1 above)
Comment (sudo-jaa) [reply framework/freedomotic-core/src/test/java/com/freedomotic/environment/GraphEdgeTest.java:206]: A classic case of @petersear :1st_place_medal:

**Review by JohnLBergqvist — APPROVED** [review]

**Review by paulkennethkent — APPROVED** [review]

**Pull request closed without merging**